packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -90,11 +90,37 @@ export default function(api, options) {
         const staticNodes = [];
         let instanceBody = [];
 
+        const ClassFieldDefinitionEvaluationTDZ = (classRef, state) => ({
+          Expression(path) {
+            if (path.parentKey === "value") {
+              path.skip();
+            }
```

Comment:
> :warning: Please don't hate me for pointing out this edge case lol

This can cause some problems. e.g.

```js
class A {
  static [{ x: A }.x];
}
```
is (correctly) compiled to
```js
class A {}

var _x$x = {
  x: (_classNameTDZError("A"), A)
}.x;
Object.defineProperty(A, _x$x, {
  configurable: true,
  enumerable: true,
  writable: true,
  value: void 0
});
```

But
```js
class A {
  static [{ x: A || 0 }.x];
}
```

is compiled to (note the missing `_classNameTDZError` call)
```js
class A {}

var _x$x = {
  x: A || 0
}.x;
Object.defineProperty(A, _x$x, {
  configurable: true,
  enumerable: true,
  writable: true,
  value: void 0
});
```

---

I suggest only traversing the computed key (`computedPath.get("key").traverse(` instead of `computedPath.traverse(`) so that this check can be avoided.

Reply:
Didnt fully investigate yet, but I tried this first! For some reason that I could not figure out it did not catch the simplest case [C], but only cases where the computed expression was more complex. It seemed like the first node was skipped.

Will follow up as soon as I get to it

Reply:
I agree but we could also do it in another PR because it would probably be simpler to review.

Reply:
Wow now these are some **real** edge cases :)

Reply:
An alternative would be to make sure that the key is `value`**and** the parentPath is the class, is that right? @nicolo-ribaudo